search/search-service.js

```diff
@@ -13,13 +14,16 @@ function createSearchService({
      * @param {string} query - The search query string.
      * @param {number} pageNumber - The page number of results to fetch.
      * @param {number} itemsPerPage - Number of items per page.
+     * @param {object} req - The request object.
      * @returns {Promise<object>} A promise that resolves to the search results.
      */
-    async function getSearchResults(query, pageNumber, itemsPerPage) {
+    async function getSearchResults(query, pageNumber, itemsPerPage, req) {
```

Comment:
Avoid passing the whole request, pass what we need.

Why is the cookie needed for the request inside `getSearchResults`?